src/go/rpk/pkg/cli/cmd/acl/create.go

```diff
@@ -11,255 +11,80 @@ package acl
 
 import (
 	"context"
-	"fmt"
-	"strings"
 
-	"github.com/Shopify/sarama"
-	"github.com/pkg/errors"
-	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
 	"github.com/spf13/cobra"
-	"golang.org/x/sync/errgroup"
+	"github.com/twmb/franz-go/pkg/kmsg"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/kafka"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/out"
 )
 
-func NewCreateACLsCommand(
-	admin func() (sarama.ClusterAdmin, error),
-) *cobra.Command {
-	var (
-		resource        string
-		resourceName    string
-		namePattern     string
-		operations      []string
-		allowPrincipals []string
-		allowHosts      []string
-		denyPrincipals  []string
-		denyHosts       []string
-	)
-	command := &cobra.Command{
-		Use:          "create",
-		Short:        "Create ACLs",
-		SilenceUsage: true,
-		Args: func(_ *cobra.Command, _ []string) error {
-			if len(allowPrincipals) == 0 &&
-				len(denyPrincipals) == 0 {
-				return fmt.Errorf(
-					"at least one of --%s or --%s must be set",
-					allowPrincipalFlag,
-					denyPrincipalFlag,
-				)
+func NewCreateCommand(fs afero.Fs) *cobra.Command {
+	var a acls
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create ACLs.",
+
+		Args: cobra.ExactArgs(0),
+		Run: func(cmd *cobra.Command, _ []string) {
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			cl, err := kafka.NewFranzClient(fs, cfg)
+			out.MaybeDie(err, "unable to initialize kafka client: %v", err)
+			defer cl.Close()
+
+			creations, err := a.createCreations()
+			out.MaybeDieErr(err)
+			if len(creations) == 0 {
+				out.Exit("Specified flags result in no ACLs to be created.")
 			}
 
-			if resource == "cluster" {
-				resourceName = defaultClusterResourceName
+			req := kmsg.NewPtrCreateACLsRequest()
+			req.Creations = creations
+			resp, err := req.RequestWith(context.Background(), cl)
+			out.MaybeDie(err, "unable to issue acl creation request: %v", err)
+			if len(resp.Results) != len(req.Creations) {
+				out.Die("response contained %d results of the expected %d!", len(resp.Results), len(req.Creations))
 			}
 
-			err := checkSupported(
-				supportedResources(),
-				resource,
-				resourceFlag,
-			)
-			if err != nil {
-				return err
+			tw := out.NewTable(headersWithError...)
+			defer tw.Flush()
+			for i, r := range resp.Results {
+				c := req.Creations[i]
```

Comment:
Is there any difference between request and response? I would lean towards printing response.

Applies to other commands (delete, list).

Reply:
The response for creations actually only contains per-creation error codes, not what was created. So, we have to lookup the request creation index to see what the error code corresponds to.

Reply:
For delete and list, we do use the response only, since the response contains everything we need.